src/phantom.js

```diff
@@ -315,8 +322,10 @@ export default class Phantom {
     }
 
     _heartBeat(): void {
-        if (this.commands.size === 0) {
-            this.execute('phantom', 'noop');
+        if (this.commands.size === 0 && !this.doing_NOOP) {
+            this.doing_NOOP = true;
+            this.process.stdin.write('NOOP' + os.EOL, 'utf8');
```

Comment:
I think the only that is required is this line and then just ignore it in `shim/index` right?